probe-rs/src/debug/source_statement.rs

```diff
@@ -57,6 +58,24 @@ impl SourceStatements {
             if row.prologue_end() {
                 prologue_completed = true;
             }
+            // For GNU C, the `DW_LNS_set_prologue_end` is not set, so we employ the same heuristic as GDB to determine when the prologue is complete.
```

Comment:
What happens for non-GCC compilers?

Reply:
The comment is misleading, and I will fix it. The test is not for GNU, but for C language standard. For any C compilers that qualify, they will either have the `DW_LNS_set_prologue_end` attribute and work like the Rust uses cases do, or they will NOT have it, and trigger this heuristic.